packages/core/src/middleware/koa-oidc-error-handler.ts

```diff
@@ -20,79 +86,39 @@ export default function koaOIDCErrorHandler<StateT, ContextT>(): Middleware<Stat
         throw error;
       }
 
-      const {
-        status = 400,
-        message,
-        error_description,
-        error_detail,
-        name,
-        expose,
-        constructor,
-        ...interpolation
-      } = error;
+      // Mimic oidc-provider's error handling, thus we can use the unified logic below.
+      // See https://github.com/panva/node-oidc-provider/blob/37d0a6cfb3c618141a44cbb904ce45659438f821/lib/shared/error_handler.js
+      ctx.status = error.statusCode || 500;
+      ctx.body = errorOut(error);
 
-      // Original OIDCProvider Error description and details are provided in the data field
-      const data = { error_description, error_detail };
-
-      switch (constructor) {
-        case errors.InvalidScope:
-        case errors.InvalidTarget:
-        case errors.InvalidToken:
-        case errors.InvalidClientMetadata:
-        case errors.InvalidRedirectUri:
-        case errors.AccessDenied:
-        case errors.UnsupportedGrantType:
-        case errors.UnsupportedResponseMode:
-        case errors.UnsupportedResponseType:
-        case errors.InvalidGrant: {
-          throw new RequestError(
-            {
-              // Manually mapped all OIDC error name to the LogtoErrorCode
-              // eslint-disable-next-line no-restricted-syntax
-              code: `oidc.${decamelize(name)}` as LogtoErrorCode,
-              status,
-              expose,
-              ...interpolation,
-            },
-            data
-          );
-        }
+      if (!EnvSet.values.isUnitTest && (!EnvSet.values.isProduction || ctx.status >= 500)) {
```

Comment:
\>= 400?

Reply:
no need to print 400<=code<500 errors in production i think. it'll be too many

Reply:
this is also the logic for `koaErrorHandler`

Reply:
ok